test/conformance/serial/ptp.go

```diff
@@ -1075,6 +1075,201 @@ spp 0
 				}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring(metrics.OpenshiftPtpThreshold),
 					"Threshold metrics are not detected")
 			})
+
+			// Verify cloud-event-proxy defaults to v2 when apiVersion is not set
+			It("Should default to event API v2 when apiVersion is not explicitly set in PtpOperatorConfig", func() {
+				By("Verifying apiVersion is not explicitly set in PtpOperatorConfig")
+				ptpOperatorConfig, err := client.Client.PtpV1Interface.PtpOperatorConfigs(pkg.PtpLinuxDaemonNamespace).Get(
+					context.Background(), pkg.PtpConfigOperatorName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				// Skip if apiVersion is explicitly set - this test is for default behavior
+				if ptpOperatorConfig.Spec.EventConfig != nil && ptpOperatorConfig.Spec.EventConfig.ApiVersion != "" {
+					Skip(fmt.Sprintf("apiVersion is explicitly set to '%s', skipping default test",
+						ptpOperatorConfig.Spec.EventConfig.ApiVersion))
+				}
+
+				By("Checking cloud-event-proxy logs for v2 API config on all pods")
+				for _, pod := range ptpPods.Items {
+					// Verify cloud-event-proxy container exists
+					cloudProxyFound := false
+					for _, c := range pod.Spec.Containers {
+						if c.Name == pkg.EventProxyContainerName {
+							cloudProxyFound = true
+						}
+					}
+					Expect(cloudProxyFound).ToNot(BeFalse(),
+						fmt.Sprintf("No cloud-event-proxy container in pod %s", pod.Name))
+
+					logrus.Infof("Checking cloud-event-proxy logs on pod %s (node: %s)", pod.Name, pod.Spec.NodeName)
+
+					// Get existing logs (no streaming)
+					podLogOptions := v1core.PodLogOptions{
+						Container: pkg.EventProxyContainerName,
+						Follow:    false,
+					}
+
+					req := client.Client.CoreV1().Pods(pod.Namespace).GetLogs(pod.Name, &podLogOptions)
+					logs, err := req.DoRaw(context.Background())
+					Expect(err).NotTo(HaveOccurred(),
+						fmt.Sprintf("Failed to get logs from pod %s", pod.Name))
+
+					logStr := string(logs)
+					apiConfigPrefix := "REST API config: version="
+					lastIndexOfApiConfigSubstringInLogs := strings.LastIndex(logStr, apiConfigPrefix)
+					if lastIndexOfApiConfigSubstringInLogs == -1 {
+						Fail(fmt.Sprintf("No 'REST API config:' found in cloud-event-proxy logs on pod %s", pod.Name))
+					}
+					// Extract the version from the last occurrence
```

Comment:
Why not use GetPodLogsRegex here instead of the code above? Same for similar code in this PR.